WIP: Update initial theme.json variables refactor

joesnellpdx opened the pull request:
### Description of the Change

Refactoring theme.json to make more agnostic - passing variables into other settings

I have the base styles connected to theme.json in via the plugin for
- default
- outline
- text
- colors for default (fill), outline - core - need colors for text and for all partials output

We still need:
- show different button colors (partial and core button) - see [.tui-content-container](https://github.com/10up/wp-scaffold-ui/pull/14) or teamwork for what is still needed
- core styles moved to plugin
- need to move custom button block to plugin (in teamwork)


Cc: @tlovett1 @devinle

joesnellpdx committed:
Update initial theme.json variables refactor

joesnellpdx committed:
Fix color palatte variables

joesnellpdx committed:
Add custom styles / output for core blocks
- theme.json
- button block custom style

joesnellpdx committed:
Add body styles to theme.json

joesnellpdx committed:
Update styles output - clean up

devinle reviewed (commented):
@joesnellpdx do you know if we should be registeringBlockStyles on the plugin side?

joesnellpdx committed:
theme.json updates

joesnellpdx committed:
Update decoration variable names. Update interactive classes in sink.

joesnellpdx committed:
remove devurl

joesnellpdx committed:
Updates to theme.json & button sink output
- add text button to sink page
- add outline and update active vars in theme.json

joesnellpdx commented:
@devinle 

All current work has been pushed.

I have the base styles connected to theme.json in via the plugin for
- default
- outline
- text
- add conditional of theme enabled to add the custom button style
- remove / provide direction for the custom file inputs targeting plugin files (in Teamwork tasks)

We still need:
- show different button colors (partial and core button)
- custom button block

I think this is a good spot to review / merge into the plugin branch before it gets too big. 

CC: @fabiankaegy

joesnellpdx committed:
Move UI Plugin core block styles to plugin

joesnellpdx committed:
Add class for vertical spacing to blocks

joesnellpdx commented:
@devinle 
@tlovett1 

This is now ready for review - works with this PR from the plugin: https://github.com/10up/wp-scaffold-ui/pull/14

joesnellpdx commented on themes/10up-theme/includes/core-block-overrides.js:
```diff
@@ -2,6 +2,12 @@
+/* TODO - ideally, block styles can be loaded conditionally as UI components allowed / added to projet */
+/* TODO - import from npm when ready */
+
+import '@10up/wp-scaffold-ui/includes/core-block-overrides';
```
@joesnellpdx 

Double check the need for this before we approve.

joesnellpdx commented on themes/10up-theme/theme.ui.json:
```diff
@@ -0,0 +1,403 @@
+          "customRadius": false
+        },
+        "color": {
+          "text": false
```
@joesnellpdx Verify this - we need text color to change outline button colors

joesnellpdx committed:
Update comment for core block overrides

fabiankaegy commented on themes/10up-theme/theme.ui.json:
```diff
@@ -0,0 +1,403 @@
+        "9-x": "calc(var(--wp--custom--space--base) * 9)",
+        "10-x": "calc(var(--wp--custom--space--base) * 10)",
+        "inset": {
+          "squish": {
```
This might be a language barear thing, but I'm not sure what this is suposed to mean / what it is going to be used for :)

joesnellpdx replied:
It is a setting to set padding on an element that is larger on the sides than on the top - like buttons, accordion headers, etc. It can get a lot more complex (see here: https://jdrf.design/tokens.html#space)... but wanted to at least have a base consistent style for it.
